docs/specs/NETWORKING.md

```diff
@@ -1,3 +1,769 @@
+- See [Ethereum specs > phase 0 > p2p interface](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/p2p-interface.md#why-are-we-using-discv5-and-not-libp2p-kademlia-dht) 
+details why discv5 was chosen over libp2p Kad DHT in Ethereum.
+
+In `v0` discv5 is used, `v1` TBD, this section will be updated once that work is complete.
+
+### Network ID
+
+Network ID is a `32byte` key, that is used to distinguish between other networks (ssv and others).
+Peers from other public/private libp2p networks (with different network ID) won't be able to read or write messages in the network, 
+meaning that the key to be known and used by all network members.
+
+It is done with [libp2p's private network](https://github.com/libp2p/specs/blob/master/pnet/Private-Networks-PSK-V1.md),
+which encrypts/decrypts all traffic with the corresponding key,
+regardless of the regular transport security protocol ([go-libp2p-noise](https://github.com/libp2p/go-libp2p-noise)).
+
+**NOTE** discovery communication (UDP) won't use the network ID, 
+and it's not needed as unknown peers will be filtered.
+
+
+### Subnets
+
+Consensus messages are being sent in the network over a subnet (pubsub topic), which the relevant peers should be subscribed to.
+
+#### Subnets - fork v0
+
+`v0` was working with a simple approach, each validator committee has 
+a dedicated pubsub topic with all the relevant peers subscribed to it (committee + exporter). \
+It helps to reduce amount of messages in the network, 
+but increases the number of topics which will grow up to the number of validators.
+
+In order to provide more redundancy, there is a global topic (AKA `main topic`) to publish all the decided messages in the network.
+
+#### Subnet - fork v1
+
+A subnet of operators is responsible for multiple committees,
+reusing the same topic to communicate on behalf of multiple validators.
+
+In comparison to `v0`, the number of topics will be reduced and the number of messages sent over the network should grow. \
+As messages will be propagated to a larger set of nodes, we can expect better reliability (arrival of messages to all operators in the committee).
+
+In addition, a larger group of operators provides:
+- redundancy of decided messages across multiple nodes
+- better security for subnets as more nodes will validate messages and can score bad/malicious nodes that will be pruned accordingly.
+
+Global/main topic is not needed in this solution and therefore will be reduced.
+
+**Validators Mapping**
+
+Validator's public key is mapped to a subnet using a hash function, 
+which helps to distribute validators across subnets in a balanced way:
+
+`hash(validatiorPubKey) % num_of_subnets`
+
+Deterministic mapping is ensured as long as the number of subnets doesn't change, 
```

Comment:
I think we should have some number simulation behind this. 
Like say SSV has 100k/ 1M validators and 128 subnets, how many msgs a second each subnet will have.
The number needs to make sense

Reply:
I'll add a table with 4/5 rows to show numbers
Calculated the best case where each operator get every message (once) for all committees in subnet:
`msgs_in_subnet = msgs_per_committee * operators_per_subnet * committees_per_subnet`